flexmeasures/data/models/reporting/__init__.py

```diff
@@ -0,0 +1,190 @@
+from typing import Optional, Union, Dict
+
+import pandas as pd
+
+from flexmeasures.data.schemas.reporting import ReporterConfigSchema
+from flexmeasures.data.models.time_series import Sensor
+from flexmeasures.data.models.data_sources import DataGeneratorMixin
+
+
+from datetime import datetime, timedelta
+
+import timely_beliefs as tb
+
+
+class Reporter(DataGeneratorMixin):
+    """Superclass for all FlexMeasures Reporters."""
+
+    __version__ = None
+    __author__ = None
+    __data_generator_base__ = "Reporter"
+
+    sensor: Sensor = None
+
+    reporter_config: Optional[dict] = None
+    reporter_config_raw: Optional[dict] = None
+    schema = ReporterConfigSchema
+    data: Dict[str, Union[tb.BeliefsDataFrame, pd.DataFrame]] = None
+    start: datetime = None
+    end: datetime = None
+    input_resolution: timedelta = None
+    belief_time: datetime = None
+
+    def __init__(
+        self, sensor: Sensor, reporter_config_raw: Optional[dict] = None
+    ) -> None:
+        """
+        Initialize a new Reporter.
+
+        Attributes:
+        :param sensor: sensor where the output of the reporter will be saved to.
+        :param reporter_config_raw: unserialized configuration of the reporter.
+        """
+
+        self.sensor = sensor
+
+        if not reporter_config_raw:
+            reporter_config_raw = {}
+
+        self.reporter_config_raw = reporter_config_raw
+
+    def fetch_data(self):
+        """
+        Fetches the time_beliefs from the database
+        """
+
+        self.data = {}
+        for tb_query in self.tb_query_config:
+
+            # using start / end instead of event_starts_after/event_ends_before when not defined
+            event_starts_after = tb_query.pop("event_starts_after", self.start)
+            event_ends_before = tb_query.pop("event_ends_before", self.end)
+            resolution = tb_query.pop("resolution", self.input_resolution)
+            belief_time = tb_query.pop("belief_time", self.belief_time)
+
+            sensor: Sensor = tb_query.pop("sensor", None)
+            alias: str = tb_query.pop("alias", None)
```

Comment:
In case we deserialized `reporter_config_raw` every time we run `compute`, it shouldn't be a problem. However, now that we deserialize just once, I made a fix for this to work. Also, I added a test to check confirm this.